src/ui/public/stringify/types/color.js

```diff
@@ -41,12 +42,22 @@ export default function ColorFormatProvider(Private) {
   };
 
   _Color.prototype._convert = {
-    html(val) {
-      const color = _.findLast(this.param('colors'), ({ range }) => {
-        if (!range) return;
-        const [start, end] = range.split(':');
-        return val >= Number(start) && val <= Number(end);
-      });
+    html(val, field) {
+
+      var color;
+      if (field.type === 'string' || field === 'string') {
```

Comment:
Since the logic for coloring numbers vs strings is pretty different (including completely different parameters), I think we should duplicate the existing formatter, and modify it to support strings instead of numbers.